src/geometry/manifold/charted_space.lean

```diff
@@ -365,6 +366,86 @@ instance : order_top (structure_groupoid H) :=
   le_top := λ u f hf, by { split; exact dec_trivial },
   ..structure_groupoid.partial_order }
 
+/-- A groupoid is closed under restriction if it contains all restrictions of its element local
+homeomorphisms to open subsets of the source. -/
+class closed_under_restriction (G : structure_groupoid H) : Prop :=
+(closed_under_restriction : ∀ {e : local_homeomorph H H}, e ∈ G.members → ∀ (s : set H), is_open s →
+  (e : local_homeomorph H H).restr s ∈ G.members)
+
+/-- The trivial restriction-closed groupoid, containing only local homeomorphisms equivalent to the
+restriction of the identity to the various open subsets. -/
+def id_restr_groupoid : structure_groupoid H :=
+{ members := {e | ∃ {s : set H} (h : is_open s), e ≈ local_homeomorph.of_set s h},
+  trans' := begin
+    rintros e e' ⟨s, hs, hse⟩ ⟨s', hs', hse'⟩,
+    refine ⟨s ∩ s', is_open_inter hs hs', _⟩,
+    have := local_homeomorph.eq_on_source.trans' hse hse',
+    rwa local_homeomorph.of_set_trans_of_set at this,
+  end,
+  symm' := begin
+    rintros e ⟨s, hs, hse⟩,
+    refine ⟨s, hs, _⟩,
+    rw [← of_set_symm],
+    exact local_homeomorph.eq_on_source.symm' hse,
+  end,
+  id_mem' := ⟨univ, is_open_univ, by simp⟩,
+  locality' := begin
+    intros e h,
+    refine ⟨e.source, e.open_source, by simp, _⟩,
+    intros x hx,
+    rcases h x hx with ⟨s, hs, hxs, s', hs', hes'⟩,
+    have hes : x ∈ (e.restr s).source,
+    { rw e.restr_source, refine ⟨hx, _⟩,
+      rw interior_eq_of_open hs, exact hxs },
+    simpa using local_homeomorph.eq_on_source.eq_on hes' hes,
```

Comment:
`simp` tends to be slow in manifold files, because there are many simp lemmas that can be used. A good tradeoff is to always use the version `simp only with mfld_simps`. Here, for instance, this would be
```suggestion
    simpa only with mfld_simps using local_homeomorph.eq_on_source.eq_on hes' hes,
```
and as you can see `simp` did not use any "exotic" lemma that is not in the simp-set `mfld_simps` designed for manifolds.

Reply:
Thanks.  I have found and replaced all the `simp`s, I think.